src/cfml/system/endpoints/Maven.cfc

```diff
@@ -0,0 +1,701 @@
+		var jarFileURL    = "";
+		var artifact      = {
+			"jarFileURL"       : "",
+			"artifactMetadata" : {}
+		};
+
+		// If the local artifact exists, serve it
+		if (
+			artifactService.artifactExists( artifactParts.artifactId, artifactParts.version ) && artifactParts.version != "STABLE" && !semanticVersion.isExactVersion(
+				artifactParts.version,
+				true
+			)
+		) {
+			job.addLog( "Lucky you, we found this version in local artifacts!" );
+			var thisArtifactPath = artifactService.getArtifactPath( artifactParts.artifactId, artifactParts.version );
+
+			// Return the path to the artifact
+			return fileEndpoint.resolvePackage(
+				thisArtifactPath,
+				currentWorkingDirectory,
+				arguments.verbose
+			);
+		}
+
+		// Check only the explicitly defined repo, if any
+		if ( len( artifactParts.repo ) ) {
+			artifact = getArtifactFromRepo(
+				artifactParts.repo,
+				artifactParts.groupId,
+				artifactParts.artifactId,
+				artifactParts.version
+			);
+			jarFileURL = artifact.jarFileURL;
+		}
+		// Otherwise, check each registered repo sequentially
+		else {
+			for ( var alias in repos ) {
+				artifact = getArtifactFromRepo(
+					repos[ alias ],
+					artifactParts.groupId,
+					artifactParts.artifactId,
+					artifactParts.version
+				);
+				jarFileURL = artifact.jarFileURL;
+				// If we found the artifact, break out of the loop
+				if ( jarFileURL.len() ) {
+					break;
+				}
+			}
+		}
+
+		// Defer to jar endpoint
+		var folderName = jarEndpoint.resolvePackage(
+			artifact.jarFileURL,
+			currentWorkingDirectory,
+			arguments.verbose
+		);
+
+		if ( artifactParts.version eq "STABLE" ) {
+			artifactParts.version = getLatestVersion( artifactParts.groupId, artifactParts.artifactId );
+		}
+
+		// Update artifact version if it's a range
+		else if ( !semanticVersion.isExactVersion( artifactParts.version, true ) ) {
+			if (
+				artifact.artifactMetadata.keyExists( "versioning" ) && artifact.artifactMetadata.versioning.keyExists( "versions" ) && artifact.artifactMetadata.versioning.versions.len()
+			) {
+				var sortedVersions = artifact.artifactMetadata.versioning.versions.sort( ( a, b ) => variables.semanticVersion.compare( b, a ) );
+				// Get the latest version that matches the range
+				for ( var thisVersion in sortedVersions ) {
+					if ( semanticVersion.satisfies( thisVersion, artifactParts.version ) ) {
+						artifactParts.version = thisVersion;
+						break;
+					}
+				}
+			}
+		}
+
+		// get dependencies
+		var artifactDependencies = getArtifactAndDependencyJarURLs(
+			artifactParts.groupId,
+			artifactParts.artifactId,
+			artifactParts.version
+		);
+
+		var installPaths = {};
+		var dependencies = {};
+
+		for ( var dependency in artifactDependencies ) {
+			if ( dependency.artifactId == artifactParts.artifactId ) {
+				continue;
+			}
+			dependencies[ dependency.artifactId ] = getNamePrefixes() & (
+				artifactParts.repo.len() ? artifactParts.repo & "|" : ""
```

Comment:
I'm not sure if the HTTP calls are processing this part of the main package's POM, but a pom.xml can register its own repo aliases to reference.  If one of the dependencies points to an alias configured in the pom.xml's `<repositories>` section, then we need to swap it out for the correct repo URL since those aliases won't be the same as what CommandBox has registered.

Reply:
This part needs some more work still. I will need to do more reading on the POM file to check for the repositories and aliases and swap the repo URL.